rpc/src/v1/helpers/subscribers.rs

```diff
@@ -16,45 +16,75 @@
 
 //! A map of subscribers.
 
-use std::ops;
+use std::{ops, str};
 use std::collections::HashMap;
 use jsonrpc_macros::pubsub::{Subscriber, Sink, SubscriptionId};
+use rand::{Rng, StdRng};
+use v1::types::H64;
 
-#[derive(Clone, Debug)]
+
+#[derive(Debug, Clone, Hash, Eq, PartialEq)]
+pub struct Id(H64);
+impl str::FromStr for Id {
+	type Err = <H64 as str::FromStr>::Err;
+
+	fn from_str(s: &str) -> Result<Self, Self::Err> {
+		Ok(Id(s.parse()?))
+	}
+}
+impl Id {
+	pub fn as_string(&self) -> String {
+		format!("{:?}", self.0)
+	}
+}
+
+#[derive(Clone)]
 pub struct Subscribers<T> {
-	next_id: u64,
-	subscriptions: HashMap<u64, T>,
+	rand: StdRng,
```

Comment:
When you run public node it would be possible to cancel other peoples subscriptions. We could instead have an id per session, but I think it would be a bit more complicated.